32blit-sdl - building for win32 failing with "undefined reference to `SDL_main'"

Using a cross-compiled SDL2 library and attempting to build a project for Win32 results in "undefined reference to `SDL_main'"

Steps to reproduce:

1. Enter any project directory
2. Make a build directory
3. Set up with `cmake .. -DCMAKE_TOOLCHAIN_FILE=../../../mingw.toolchain -DSDL2_DIR=/usr/local/cross-tools/i686-w64-mingw32/lib/cmake/SDL2 -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON`
4. Run "make"

Example verbose output:

```
[100%] Linking CXX executable platformer.exe
/usr/bin/cmake -E cmake_link_script CMakeFiles/platformer.dir/link.txt --verbose=1
/usr/bin/cmake -E remove -f CMakeFiles/platformer.dir/objects.a
/usr/bin/i686-w64-mingw32-ar cr CMakeFiles/platformer.dir/objects.a @CMakeFiles/platformer.dir/objects1.rsp
/usr/bin/i686-w64-mingw32-g++    -Wl,--whole-archive CMakeFiles/platformer.dir/objects.a -Wl,--no-whole-archive  -o platformer.exe -Wl,--out-implib,libplatformer.dll.a -Wl,--major-image-version,0,--minor-image-version,0 @CMakeFiles/platformer.dir/linklibs.rsp
/usr/local/cross-tools/i686-w64-mingw32/lib/libSDL2main.a(SDL_windows_main.o): In function `main_getcmdline':
/mnt/c/Users/32blit/Documents/Development/Pimoroni/SDL2-2.0.10/build.mingwx86/../src/main/windows/SDL_windows_main.c:175: undefined reference to `SDL_main'
collect2: error: ld returned 1 exit status
CMakeFiles/platformer.dir/build.make:102: recipe for target 'platformer.exe' failed
make[2]: *** [platformer.exe] Error 1
make[2]: Leaving directory '/mnt/c/Users/32blit/Documents/Development/Pimoroni/32blit-beta/examples/platformer/build.mingw'
CMakeFiles/Makefile2:71: recipe for target 'CMakeFiles/platformer.dir/all' failed
make[1]: *** [CMakeFiles/platformer.dir/all] Error 2
make[1]: Leaving directory '/mnt/c/Users/32blit/Documents/Development/Pimoroni/32blit-beta/examples/platformer/build.mingw'
Makefile:86: recipe for target 'all' failed
make: *** [all] Error 2
```

Main exists in `Main.cpp` of the SDL HAL, with the correct signature.

--- a/32blit-sdl/CMakeLists.txt
+++ b/32blit-sdl/CMakeLists.txt
@@ -50,5 +50,5 @@ endif()
 
 function(blit_executable NAME SOURCES)
 	add_executable(${NAME} ${SOURCES} ${ARGN})
-  	target_link_libraries(${NAME} BlitHalSDL)
+  	target_link_libraries(${NAME} -Wl,--whole-archive BlitHalSDL -Wl,--no-whole-archive)
 endfunction()

--- a/32blit-sdl/Main.hpp
+++ b/32blit-sdl/Main.hpp
@@ -1,4 +1,0 @@
-#pragma once
-#include "windows.h"
-int main();
-int WINAPI wWinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, PWSTR pCmdLine, int nCmdShow);

--- a/32blit-sdl/Renderer.cpp
+++ b/32blit-sdl/Renderer.cpp
@@ -1,4 +1,5 @@
 #include <algorithm>
+#include <cstdio>
 #include <SDL.h>
 
 #include "Renderer.hpp"

